### DISCUSSION TITLE
consuming from mediamtx whep endpoint using gstreamer whepsrc with user and password auth

### DISCUSSION THREAD

**Discussion by olivetom** [bluenviron/mediamtx#3732] [Questions] [answered]
### Question

Hopefully someone can give me a hand on how to consume from mediamtx using gstreamer whepsrc using basic user and password authentication. I've tryed using whep-endpoint=http://user:password@server:8889/mystream/ and whep-endpoint=http://server:8889/mystream?user=user&pass=password but the failed with message Unexpected response: 404 - 404 page not found

Despite this gstreamer playback issue, I'm able to do playback from chrome smoothly.

Thanks in advance

**Comment by alim989898** [comment]
rtsp: yes
rtspAddress: :8554
hls: yes
hlsAddress: :8888
authMethod: internal
authInternalUsers:
- user: "alim"
  pass: "password123456789"
  ips: []
  permissions:
  - action: publish
    path:
  - action: read
    path:
  - action: playback
    path:  
    
paths:
  stram1:
    source: rtsp://alim:passswe!@192.168.1.99:554/ISAPI/Streaming/Channels/101


vlc open 
http://ip:8888/stram1/index.m3u8
rtsp://alim:password123456789@ip:8554/stram1

**Reply by olivetom** [reply]
@alim989898 rtsp for me works okay as well. I'm trying with whep.

**Comment by aler9** [comment, marked as answer]
Hello, as written in the README, the WHEP endpoint needs a "/whep" suffix, therefore:

```
whep-endpoint=http://user:password@server:8889/mystream/whep
```

**Reply by olivetom** [reply]
Thanks for letting me know about that overlooked detail.

[a comment by an automated account is not shown]
